### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by pickjasmine**
Completes the more important parts of #134 

* Combines resource and policy state into one application state
* Updates the test renderer to handle changes in state so you can use the `rerender` utility when state needs to change between tests.
* Fixes a weird behavior when you were in the policy playground searching for a policy or resource and pressed the search button without a search term. Previously would lead to a never ending loading spinner until you typed into the search bar or clicked the "View all" button, but now it defaults to showing all policies/resources

Will follow up with the refactoring of CSS in another PR because this one is getting lengthy.

### CODE REVIEW

File: components/policies/PolicyForm.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ const PolicyForm = ({
     const policy = await response.json();
 
     dispatch({
```
Comment (alexashley) [inline components/policies/PolicyForm.js:121]: Not sure why exactly, but it seems like the policy version history isn't updating after a new version is made.

File: test/reducers/appState.spec.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        data,
+      });
+
+      expect(actual).toBeDefined();
+    });
+
+    expect.assertions(actions.length);
+  });
+
+  it("should update state when the action type is 'SET_POLICY_SEARCH_TERM'", () => {
```
Comment (alexashley) [inline test/reducers/appState.spec.js:43]: Might be a good spot for a [table test](https://jestjs.io/docs/api#testeachtablename-fn-timeout):

```javascript
  it.each([
    {
      action: stateActions.SET_POLICY_SEARCH_TERM,
      fieldName: 'policySearchTerm'
    }
  ])("should update state when the action type is '$action'", ({action, fieldName}) => {
    const actual = appStateReducer(state, {
      type: action,
      data,
    });

    expect(actual).toEqual({
      ...state,
      [fieldName]: data,
    });
  })
```

File: components/policies/PolicyForm.js (hunk 1 above)
Comment (pickjasmine) [reply components/policies/PolicyForm.js:121]: Hmmmm.... I am not able to reproduce. Could you show me?

File: test/reducers/appState.spec.js (hunk 2 above)
Comment (pickjasmine) [reply test/reducers/appState.spec.js:43]: Amazing, I did not know this existed. I'll update this.

File: components/policies/PolicyForm.js (hunk 1 above)
Comment (alexashley) [reply components/policies/PolicyForm.js:121]: Never mind me, I could only get it to happen with a single policy and can't reproduce it after nuking `.next`.

**Review by alexashley — APPROVED** [review]

**Pull request merged**